crates/turborepo-scm/src/package_deps.rs

```diff
@@ -99,11 +116,52 @@ impl Git {
         turbo_root: &AbsoluteSystemPath,
         package_path: &AnchoredSystemPath,
         inputs: &[S],
+        include_default_files: bool,
     ) -> Result<GitHashes, Error> {
+        if include_default_files && !inputs.is_empty() {
```

Comment:
Probably a matter of preference, but could we rearrange the conditions in this fn:

```rust
// If inputs isn't specified, get everything
if inputs.is_empty() {
  return self.get_package_file_hashes_from_index(turbo_root, package_path)
}

// we have inputs, and we don't need the default set
if !include_default_files {
  return self.get_package_file_hashes_from_inputs(turbo_root, package_path, inputs, true)
}

// we have inputs and we want all the defaults, we have to merge includes/excludes
// new logic here
```